kymatio/scattering1d/backend/torch_skcuda_backend.py

```diff
@@ -149,13 +147,10 @@ def __call__(self, x, k):
         if not x.is_cuda and self.backend == 'skcuda':
             raise TypeError('Use the torch backend (without skcuda) for CPU tensors.')
 
-        if not _is_complex(x):
-            raise TypeError('The input and outputs should be complex.')
-
-        if not x.is_contiguous():
-            raise RuntimeError('Input should be contiguous.')
+        contiguous_check(x) 
+        complex_check(x)
 
-        out = x.new(x.shape[0], x.shape[1], x.shape[2] // k, 2)
+        out = torch.empty_like(x[:, :, ::k])
```

Comment:
why?

Reply:
Same reason as before: it obscures the fact that we simply want to create an array with the same dtype, layout, and device as `x`, but with a different shape. We're creating an intermediate object `x[:, :, ::k]` just to get its shape, which is a bit of misdirection. In this case, it probably doesn't cause much overhead, since the new array is just a view of the old one, but it's not always the case.

Reply:
there was a similar one that you fixed yesterday.

I do prefer the shorter version, but the longer version is more explicit, though less readable.

I guess the fear is that `x.shape[-1]` might not be divisble by `k` and the slicing one would not cause an error. The hope is also that `empty_like` doesn't copy stride pattern ... (that would be verifiable)

Reply:
oh our answers overlapped.

I don't think that slicing an array causes more a lot of overhead. It does create an array descriptor, but that pretty much just contains shape dtype etc, which are given explicitly in the replacement command.
Seems pretty much the same, but the slicing one is a fair bit more readable in mine eyes

Reply:
I'm fine with either solution

Reply:
ok ill do it.

Reply:
if I do this i get CUDA error: an illegal memory access was encountered

Reply:
wait, i think you gave me the wrong shape.

Reply:
Um… that's weird.

Reply:
I can fix it no worries.

Reply:
Where is that error triggered?